crates/rust/src/lib.rs

```diff
@@ -1049,9 +1080,15 @@ impl WorldGenerator for RustWasm {
         let module_name = name.to_snake_case();
         files.push(&format!("{module_name}.rs"), src.as_bytes());
 
-        let remapping_keys = self.with.keys().cloned().collect::<HashSet<String>>();
+        let remapped_keys = self
+            .with
+            .iter()
+            .filter(|(_, r)| matches!(r, InterfaceGeneration::Remap(_)))
```

Comment:
Could this work by removing `InterfaceGeneration::Generate` throughout generating bindings? That way erroneous `"foo": generate` keys in `with` could still be warned against

Reply:
Interfaces can be encountered more than once, so we don't want to remove the `InterfaceGeneration::Generate` as we go as we will then fail to generate later references to that interface.

I believe I've found a solution that works though.